### PR TITLE
June13 set of stack

### PR DESCRIPTION
**Pull request by VivianWenhuiLiang**

### CODE REVIEW

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  claritification: s2 is a rotation of s1,must s1.len=s2.len
+                  (rotation: move like clockwise)
+ approach:s1.len==s2.len,if not return false
+          S2 must is substring of S1+S1
+ complexcity: o(n),momery is O(n)
+ */
+public class CC1_8_isRotation {
+    public boolean isRotation(String s1, String s2) {
+        if(s1.length() == s2.length() && s1.length() > 0) {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/CC1_8_isRotation.java]: Null check
```
if (s1 == null || s2 == null || s1.length() != s2.length()) {
    return false;
}
```

Also, could you clarify what should be returned if both s1 and s2 are empty?

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            String ss1 = s1 + s1;
+            return isSubstring(ss1, s2);
+        }
+        return false;
+    }
+
+    public boolean isSubstring(String s1, String s2) {
+        if(s1.length() < s2.length() || s1.length() == 0 || s2.length() == 0) {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/CC1_8_isRotation.java]: Please comment why return false if both s1 and s2 are empty?

File: src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+approach: use three pointer: cur(iterator for grand loop; the end edge of inner loop)
+                             pre(maker the current non_duplicate element)
+                             run(inner loop, find the duplicate)
+
+complexity: runtime complexity o(n*n), memory complexity o(1) only for the variables
+*/
+
+public class CCI2_1RemoveDupsWithoutBuffer {
+    public static class Node<T> {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java]: I thought you have moved Node out to it's own file. Can you use that instead?

File: src/com/vivian/java/leetcode/CCI_5FindLoopStart.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                                                   because step2 is 2 times peed of step1,so
+                                                   2(A+C)=A+C+nB => A=(n-1)B+B-C 
+                                                   =>A=B-C (this is why head to beginng diatance == meeting point to beginning)
+
+
+complexity: runtime complexity O(n),worse case step1 visvited every element in the linkedlist,and step2 2n
+            memory complexit O(1)
+            */
+
+public class CCI_5FindLoopStart {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/CCI_5FindLoopStart.java]: What's the difference between this and CCI2_5FindLoopStart.java?

File: src/com/vivian/java/leetcode/Q171_ExcelSheetColumnNumber.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.vivian.java.leetcode;
+
+public class Solution {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/Q171_ExcelSheetColumnNumber.java]: Please comment Clarification, Approach, Edge Case and Complexity.

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java (hunk 1 above)
Comment (VivianWenhuiLiang) [reply src/com/vivian/java/leetcode/CC1_8_isRotation.java]: if (s1 == null || s2 == null || s1.length() != s2.length()) {
    return false;
}                    

where are this code?

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java (hunk 2 above)
Comment (VivianWenhuiLiang) [reply src/com/vivian/java/leetcode/CC1_8_isRotation.java]: ok ,I throw an llegalArgumentException

File: src/com/vivian/java/leetcode/CCI_5FindLoopStart.java (hunk 4 above)
Comment (VivianWenhuiLiang) [reply src/com/vivian/java/leetcode/CCI_5FindLoopStart.java]: I wasn't rename successively

File: src/com/vivian/java/leetcode/Q171_ExcelSheetColumnNumber.java (hunk 5 above)
Comment (VivianWenhuiLiang) [reply src/com/vivian/java/leetcode/Q171_ExcelSheetColumnNumber.java]: that's your contribution,but I will do it what you commit above

File: src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java (hunk 3 above)
Comment (VivianWenhuiLiang) [reply src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java]: the faction is remove repeated node out its own file

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java (hunk 1 above)
Comment (GeorgeZhiXu) [reply src/com/vivian/java/leetcode/CC1_8_isRotation.java]: I’m saying you should add this bull check at this line.

File: src/com/vivian/java/leetcode/CC1_8_isRotation.java (hunk 2 above)
Comment (GeorgeZhiXu) [reply src/com/vivian/java/leetcode/CC1_8_isRotation.java]: Ok. Or should you return true? 
I’m not saying you should, but it’s a good clarification question to ask.

File: src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java (hunk 3 above)
Comment (GeorgeZhiXu) [reply src/com/vivian/java/leetcode/CCI2_1RemoveDupsWithoutBuffer.java]: Not sure I follow.

**Pull request merged**